lib/all_your_base_cases.rb

```diff
@@ -0,0 +1,106 @@
+class AllYourBaseCase < OpenStruct
+  def test_name
+    'test_%s' % description.downcase.tr(' -', '_')
+  end
+
+  def workload
+    (
+      [
+        "digits = #{input_digits}",
+        indent(4, "input_base = #{input_base}"),
+        indent(4, "output_base = #{output_base}"),
+      ] + assertion
+    ).join("\n")
+  end
+
+  def skipped
+    index.zero? ? '# skip' : 'skip'
+  end
+
+  private
+
+  def indent(size, lines)
+    lines.lines.each_with_object('') { |line, obj| obj << ' ' * size + line }
+  end
+
+  def assertion
+    return error_assertion unless expected
+
+    [
+      indent(4, "expected = #{expected}"),
+      "",
+      indent(4, "converted = BaseConverter.convert(input_base, digits, output_base)"),
+      "",
+      indent(4, "assert_equal expected, converted,"),
+      indent(17, error_message)
+    ]
+  end
+
+  def error_message
+    [
+      '"Input base: #{input_base}, output base #{output_base}. " \\',
+      '"Expected #{expected} but got #{converted}."',
+    ].join("\n")
+  end
+
+  def error_assertion
+    [
+      "",
+      indent(4, "assert_raises ArgumentError do"),
+      indent(6, "BaseConverter.convert(input_base, digits, output_base)"),
+      indent(4, "end"),
```

Comment:
One thing that bugs me about this otherwise great solution\* is all the micro-management of indentation that is going on.
There must be a more elegant way of handling it.

\* not just in this method, but throughout the whole file.